### PR TITLE
DATAGO-88864 - Add annotation and improve test's on 2 python versions

### PR DESCRIPTION
**Pull request by artyom-morozov**
[DATAGO-88864](https://sol-jira.atlassian.net/browse/DATAGO-88864)

### What is the purpose of this change?

- Imporvements for for setup hatch to make it work better with 2 python versions 
- New pr comment to show test results and coverage on a PR 

Screenshot 2024-11-20 at 2 16 25 PM

### How is this accomplished?

- Rely on hatch to use pyproject.toml for installing dependencies
- Handle a case where there ios a discrepancy between dependencies in pyproject.toml and requirements.txt files 

Example Run 

https://github.com/SolaceDev/solace-ai-connector/pull/64

[DATAGO-88864]: https://sol-jira.atlassian.net/browse/DATAGO-88864?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/actions/hatch-setup/action.yml [hunk 1]
```diff
@@ -25,37 +28,65 @@ runs:
...
-      if: steps.cache-restore.outputs.cache-hit != 'true'
-      id: cache-hatch
-      with:
-        path: |
-          ${{ env.HATCH_CACHE_DIR }}
-          ${{ env.HATCH_DATA_DIR }}
-        key: ${{ runner.os }}-hatch-${{ hashFiles('pyproject.toml') }}
+    - name: Install Requirements.txt Dependencies for default python version
+      shell: bash
+      if: hashFiles('requirements.txt') != '' && steps.cache.outputs.cache-hit != 'true' && steps.test-matrix-present.outputs.matrix-present == 'false'
```
Comment (johnvincentcorpuz) [inline .github/actions/hatch-setup/action.yml:77]: Not following why `matrix-present` has to be false here.

Comment (artyom-morozov) [reply .github/actions/hatch-setup/action.yml:77]: For matrix present we basicaly verify if the following settings are present in the pyproject.toml file.
```
[tool.hatch.envs.hatch-test]
installer = "pip"

# # Specify minimum and maximum Python versions to test
[[tool.hatch.envs.hatch-test.matrix]]
python = ["3.10", "3.12"]
```
Those settings allow hatch to use 2 separate environments with different versions of python and use preinstalled test dependencies. 
Without it we can only test reliably on one version of python.
So depending on if we have matrix setup in the settings or not. We need to call hatch differently.

File: .github/workflows/hatch_ci.yml [hunk 2]
```diff
@@ -36,7 +38,8 @@ jobs:
           fetch-depth: 0
 
       - name: Set up Hatch
-        uses: SolaceDev/solace-public-workflows/.github/actions/hatch-setup@v1.0.0
+        id: hatch-setup
+        uses: SolaceDev/solace-public-workflows/.github/actions/hatch-setup@test
```
[the remark this answers is not in the document]
Comment (artyom-morozov) [reply .github/workflows/hatch_ci.yml]: Fixed to use `@latest` tag

**Review by AmanRiat1 — APPROVED** [review]
lgtm!

**Pull request merged**